src/java/org/apache/cassandra/service/accord/serializers/IVersionedWithKeysSerializer.java

```diff
@@ -171,6 +171,36 @@ else if (supersetCount < 64)
             }
         }
 
+        @DontInline
+        private <K extends Routable, R extends Routables<K>> long serializeLargeSubsetSize(R serialize, int serializeCount, R superset, int supersetCount)
```

Comment:
i tried to go with a simpler run length encoding, and this actually takes up less space than the original!

The ideas are the same, but I just tried to simplify things

1) the index isn't a delta, its an absolute
2) take is the same, I mostly just rename it.  It did have an off-by-one bug before